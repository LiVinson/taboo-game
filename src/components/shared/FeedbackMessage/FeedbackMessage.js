import React from "react"
import PropTypes from "prop-types"
import {StyledErrorMessage, StyledSuccessMessage} from "./style"

export const ErrorMessage = ({error, large, light}) => {
    return <StyledErrorMessage light={light} large={large}>{error}</StyledErrorMessage>
}

ErrorMessage.propTypes = {
    error: PropTypes.string,
    large: PropTypes.bool,
    light: PropTypes.bool
}

ErrorMessage.defaultProp = {
    large: false,
    error: "A problem has occured. Please try again."
}

export const SuccessMessage = ({message}) => {
    return <StyledSuccessMessage>{message}</StyledSuccessMessage>
}



SuccessMessage.propTypes = {
    message: PropTypes.string.isRequired,
}
