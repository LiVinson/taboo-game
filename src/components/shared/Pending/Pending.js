import React from 'react'
import PropTypes from 'prop-types'
import { StyledPending, StyledLargePending } from './style'

export class Pending extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			message: this.props.message,
		}
	}

	componentDidMount() {
		const { speed, message } = this.props

		this.setPendingMessage(message, speed)
	}

	setPendingMessage = (message, speed) => {
		this.interval = window.setInterval(() => {
			this.state.message === message + '...'
				? this.setState({ message: message })
				: this.setState(({ message }) => ({ message: message + '.' }))
		}, speed)
	}
	componentDidUpdate(prevProps) {
		if (prevProps.message !== this.props.message) {
			window.clearInterval(this.interval)
			this.setState(
				{
					message: this.props.message,
				},
				this.setPendingMessage(this.props.message, this.props.speed)
			)
		}
	}
	componentWillUnmount() {
		window.clearInterval(this.interval)
	}

	render() {
		const { large } = this.props
		const pending = large ? (
			<StyledLargePending>{this.state.message}</StyledLargePending>
		) : (
			<StyledPending>{this.state.message}</StyledPending>
		)
		return pending
	}
}

Pending.defaultProps = {
	speed: 300,
	large: false,
}

Pending.propTypes = {
	message: PropTypes.string.isRequired,
	speed: PropTypes.number.isRequired,
	large: PropTypes.bool.isRequired,
}
