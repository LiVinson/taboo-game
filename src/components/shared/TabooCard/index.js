		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
import { TabooCard, ButtonTabooCard, LayeredTabooCard, TabooCardTop, FilteredTabooList} from "./TabooCard"

		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
export { TabooCard, ButtonTabooCard,  LayeredTabooCard, TabooCardTop, FilteredTabooList};

