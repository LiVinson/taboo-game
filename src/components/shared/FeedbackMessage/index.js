import { ErrorMessage, SuccessMessage } from "./FeedbackMessage"

export {ErrorMessage, SuccessMessage}