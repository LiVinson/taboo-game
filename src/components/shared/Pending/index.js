import {Pending} from "./Pending"

export default Pending