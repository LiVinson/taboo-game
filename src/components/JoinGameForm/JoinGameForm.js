import React from 'react'
import PropTypes from 'prop-types'
import { Formik, Field, ErrorMessage } from 'formik'
import { Form, FormSection, TextInput, TextLabel, ErrorText } from 'components/shared/FormElements'

const validate = (values) => {
	const errors = {}
	if (!values.name) {
		errors.name = 'Name is Required'
	} else if (values.name.length > 25) {
		errors.name = 'Must be 25 characters or less'
	}

	if (!values.gamecode) {
		errors.gamecode = 'Game code is required'
	}
	return errors
}

const JoinGameForm = (props) => {
	return (
		<Formik
			initialValues={props.initialValues}
			validate={validate}
			onSubmit={(values, { setSubmitting }) => props.handleSubmit(values, setSubmitting)}
		>
			{(formik) => (
				<Form id="joinGameForm">
					<FormSection>
						<TextLabel htmlFor="name">Enter Your Name</TextLabel>
						<Field id="name" name="name" type="text" as={TextInput} placeholder="" tabIndex="1" />
						<ErrorMessage name="name" component={ErrorText} />
					</FormSection>
					<FormSection>
						<TextLabel htmlFor="gamecode">Enter a Game Code</TextLabel>
						<Field id="gamecode" name="gamecode" type="text" as={TextInput} placeholder="" tabIndex="1" />
						<ErrorMessage name="gamecode" component={ErrorText} />
					</FormSection>
				</Form>
			)}
		</Formik>
	)
}

JoinGameForm.propTypes = {
	initialValues: PropTypes.object.isRequired,
}

export default JoinGameForm
