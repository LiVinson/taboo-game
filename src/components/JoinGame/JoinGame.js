import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import { ButtonTabooCard } from 'components/shared/TabooCard'
import JoinGameForm from 'components/JoinGameForm'
import Pending from 'components/shared/Pending'
import {ErrorMessage} from 'components/shared/FeedbackMessage'
import { joinNewGame } from 'store/actions/gameActions'
import { clearErrors } from 'store/actions/errorActions'

export class JoinGame extends React.Component {
	constructor(props) {
		super(props)

		this.state = {
			name: '',
			gamecode: '',
			redirect: false,
		}
	}

	//Update to creating player id, validating gamecode, redirecting to 'Waiting Room'
	handleSubmit = (values, setSubmitting) => {
		this.setState(
			{
				playerName: values.name.toUpperCase(),
				gamecode: values.gamecode.toUpperCase(),
			},
			() => {
				const { gamecode, playerName } = this.state
				this.props.joinNewGame({ gamecode, playerName }).then(() => {
					//Finishes Formik submission process
					setSubmitting(false)
					//redirect to Waiting
					this.setState({
						redirect: true,
					})
				})
			}
		)
	}

	handleBackClick = () => {
		this.props.history.push('/home')
	}

	//Needed in case user gets game error and opens component that references same error property without clearing
	componentWillUnmount() {
		this.props.clearGameErrors()
	}

	render() {
		const { name, gamecode } = this.state
		const buttonInfo = [
			{ text: 'Back', onClick: this.handleBackClick },
			{
				form: 'joinGameForm',
				text: 'Join',
				type: 'submit',
				disabled: this.props.isPending,
			},
		]

		return this.state.redirect ? (
			<Redirect to={`/waiting/${this.state.gamecode}`} />
		) : (
			<ButtonTabooCard tabooWord="Join Game" buttons={buttonInfo}>
				<JoinGameForm initialValues={{ name, gamecode }} handleSubmit={this.handleSubmit} />
				{this.props.isPending ? <Pending speed={300} message={`Joining game ${this.state.gamecode}`} /> : null}
				{this.props.error ? <ErrorMessage error={this.props.error} /> : null}
			</ButtonTabooCard>
		)
	}
}

JoinGame.propTypes = {
	history: PropTypes.object.isRequired,
	error: PropTypes.string,
	isPending: PropTypes.bool.isRequired,
	joinNewGame: PropTypes.func.isRequired,
	clearGameErrors: PropTypes.func.isRequired,
}

const mapStateToProps = (state) => {
	// console.log(state)
	return {
		error: state.game.error ? state.game.error.errorMessage : state.game.error,
		isPending: state.game.pending,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		joinNewGame: (gameData) => dispatch(joinNewGame(gameData)),
		clearGameErrors: () => dispatch(clearErrors("CLEAR_GAME_ERRORS"))
	}
}
export default connect(mapStateToProps, mapDispatchToProps)(JoinGame)
