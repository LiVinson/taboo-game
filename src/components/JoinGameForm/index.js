import JoinGameForm  from "./JoinGameForm"

export default JoinGameForm