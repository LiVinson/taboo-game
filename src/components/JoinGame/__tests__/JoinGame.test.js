import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import {JoinGame} from '../JoinGame'
import { ButtonTabooCard } from 'components/shared/TabooCard'
import JoinGameForm from 'components/JoinGameForm'
import Pending from 'components/shared/Pending'
import {ErrorMessage} from 'components/shared/FeedbackMessage'

describe('JoinGame functionality and rendering', () => {
	const defaultProps = {
		history: {
			push: jest.fn(),
		},
		error: null,
		isPending: false,
		gamecode: '',
		joinNewGame: jest.fn(() => Promise.resolve()),
		clearGameErrors: jest.fn(),
	}

	const formValues = {
		name: 'Sam',
		gamecode: 'code12',
	}

	test('JoinGame renders without crashing', () => {
		const props = {
			...defaultProps,
		}
		shallow(<JoinGame {...props} />)
	})

	/* ---------------------SNAPSHOT TESTS & RENDERING ------------------------------- */

	test('JoinGame renders correctly with default props', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<JoinGame {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('JoinGame renders correctly with isPending props true', () => {
		const props = {
			...defaultProps,
			isPending: true,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<JoinGame {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('JoinGame renders correctly with error props true', () => {
		const props = {
			...defaultProps,
			error: 'There was an error joining the game.',
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<JoinGame {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})
	test('JoinGame renders a ButtonTabooCard with a Form', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = shallow(
			<ThemeProvider theme={theme}>
				<JoinGame {...props} />
			</ThemeProvider>
		)
		//Need to use find to access JoinGame due to ThemeProvider wrapper and use .dive() for child components due to shallow rendering
		const component = wrapper.find(JoinGame)
		expect(component.dive().find(ButtonTabooCard).length).toBe(1)
		expect(component.dive().find(JoinGameForm).length).toBe(1)

	})

	/* --------------------- FUNCTIONALITY & PROPS ------------------------------- */

	test('JoinGame navigates back to /home when handleBackClick is called', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<JoinGame {...props} />)

		wrapper.instance().handleBackClick()
		expect(props.history.push).toHaveBeenCalled()
		expect(props.history.push).toHaveBeenCalledWith('/home')
	})

	test('Calls props.joinNewGame with form data on submit', () => {
		const props = {
			...defaultProps,
		}

		const setSubmittingMock = jest.fn()
		const wrapper = shallow(<JoinGame {...props} />)

		wrapper.instance().handleSubmit(formValues, setSubmittingMock)
		//setTimout gives time for the React.setState to run and it's callback before running test
		setTimeout(() => {
			const { endGameMethod, endValue, skipPenalty, name } = wrapper.state()
			const gameData = {
				status: 'new',
				endGameMethod,
				endValue,
				skipPenalty,
				players: [],
			}
			expect(props.joinNewGame).toHaveBeenCalled()
			expect(props.joinNewGame).toHaveBeenCalledWith('CODE00', gameData, name)
		}, 75)
	})

	test('Does not render Pending component when props.isPending is false', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<JoinGame {...props} />)
		expect(wrapper.find(Pending)).toHaveLength(0)
	})

	test('renders Pending component when props.isPending is true', () => {
		const props = {
			...defaultProps,
			isPending: true,
		}
		const wrapper = shallow(<JoinGame {...props} />)
		expect(wrapper.find(Pending)).toHaveLength(1)
	})

	test('Does not render ErrorMessage component when props.isPending is false', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<JoinGame {...props} />)
		expect(wrapper.find(ErrorMessage)).toHaveLength(0)
	})

	test('renders ErrorMessage component when props.isPending is true', () => {
		const props = {
			...defaultProps,
			error: 'There was an error joining the game.',
		}
		const wrapper = shallow(<JoinGame {...props} />)
		expect(wrapper.find(ErrorMessage)).toHaveLength(1)
	})

	test('JoinGame calls props.clearGameErrors on unmount', () => {
		const props = {
			...defaultProps,
			error: 'There was an error joining the game.',
		}

		//Requires mount for access to lifecycle methods
		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<JoinGame {...props} />
			</ThemeProvider>
		)
		wrapper.unmount()
		expect(props.clearGameErrors).toHaveBeenCalled()
	})
})
