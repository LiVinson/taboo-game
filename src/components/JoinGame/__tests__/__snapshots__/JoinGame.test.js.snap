// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinGame functionality and rendering JoinGame renders correctly with default props 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c8 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c8:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c8:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c8:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c8:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c9 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c9:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c9:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c9:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c9:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c9:focus {
  outline: solid 2px #f7f9f9;
}

.c7 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  text-align: left;
  font-size: 2.2rem;
  font-family: 'Oswald',sans-serif;
  color: #333333;
}

.c4 {
  margin-bottom: 1.5rem;
  padding-left: 0;
  padding-right: 0;
}

.c6 {
  width: 100%;
  padding: 1rem;
  border: none;
  background-color: #ebebeb;
  font-size: 2.2rem;
  border-radius: 2%;
  text-align: center;
  text-transform: uppercase;
}

.c6::-webkit-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::-moz-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:focus {
  outline: none;
  border: solid 1px #1982C4;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  font-size: 1.8;
  width: 100%;
  margin-bottom: 0.8rem;
  text-align: center;
  font-weight: 500;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Join Game
  </div>
  <div
    className="c2"
  >
    <form
      action="#"
      className="c3"
      id="joinGameForm"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="c4"
      >
        <label
          className="sc-fznxsB c5"
          htmlFor="name"
        >
          Enter Your Name
        </label>
        <input
          className="c6"
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          tabIndex="1"
          type="text"
          value=""
        />
      </div>
      <div
        className="c4"
      >
        <label
          className="sc-fznxsB c5"
          htmlFor="gamecode"
        >
          Enter a Game Code
        </label>
        <input
          className="c6"
          id="gamecode"
          name="gamecode"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          tabIndex="1"
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
  <div
    className="c7"
  >
    <button
      className="c8"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="c9"
      disabled={false}
      form="joinGameForm"
      type="submit"
    >
      Join
    </button>
  </div>
</div>
`;

exports[`JoinGame functionality and rendering JoinGame renders correctly with error props true 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c9 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c9:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c9:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c9:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c9:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c10 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c10:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c10:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c10:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c10:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c10:focus {
  outline: solid 2px #f7f9f9;
}

.c8 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c7 {
  font-size: 1.5rem;
  font-style: italic;
  color: red;
}

.c3 {
  text-align: left;
  font-size: 2.2rem;
  font-family: 'Oswald',sans-serif;
  color: #333333;
}

.c4 {
  margin-bottom: 1.5rem;
  padding-left: 0;
  padding-right: 0;
}

.c6 {
  width: 100%;
  padding: 1rem;
  border: none;
  background-color: #ebebeb;
  font-size: 2.2rem;
  border-radius: 2%;
  text-align: center;
  text-transform: uppercase;
}

.c6::-webkit-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::-moz-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:focus {
  outline: none;
  border: solid 1px #1982C4;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  font-size: 1.8;
  width: 100%;
  margin-bottom: 0.8rem;
  text-align: center;
  font-weight: 500;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Join Game
  </div>
  <div
    className="c2"
  >
    <form
      action="#"
      className="c3"
      id="joinGameForm"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="c4"
      >
        <label
          className="sc-fznxsB c5"
          htmlFor="name"
        >
          Enter Your Name
        </label>
        <input
          className="c6"
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          tabIndex="1"
          type="text"
          value=""
        />
      </div>
      <div
        className="c4"
      >
        <label
          className="sc-fznxsB c5"
          htmlFor="gamecode"
        >
          Enter a Game Code
        </label>
        <input
          className="c6"
          id="gamecode"
          name="gamecode"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          tabIndex="1"
          type="text"
          value=""
        />
      </div>
    </form>
    <p
      className="c7"
    >
      There was an error joining the game.
    </p>
  </div>
  <div
    className="c8"
  >
    <button
      className="c9"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="c10"
      disabled={false}
      form="joinGameForm"
      type="submit"
    >
      Join
    </button>
  </div>
</div>
`;

exports[`JoinGame functionality and rendering JoinGame renders correctly with isPending props true 1`] = `
.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c9 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c9:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c9:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c9:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c9:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c10 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c10:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c10:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c10:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c10:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c10:focus {
  outline: solid 2px #f7f9f9;
}

.c8 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  text-align: left;
  font-size: 2.2rem;
  font-family: 'Oswald',sans-serif;
  color: #333333;
}

.c4 {
  margin-bottom: 1.5rem;
  padding-left: 0;
  padding-right: 0;
}

.c6 {
  width: 100%;
  padding: 1rem;
  border: none;
  background-color: #ebebeb;
  font-size: 2.2rem;
  border-radius: 2%;
  text-align: center;
  text-transform: uppercase;
}

.c6::-webkit-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::-moz-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:focus {
  outline: none;
  border: solid 1px #1982C4;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  font-size: 1.8;
  width: 100%;
  margin-bottom: 0.8rem;
  text-align: center;
  font-weight: 500;
}

.c7 {
  font-size: 1.5rem;
  font-style: italic;
  color: #1982C4;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Join Game
  </div>
  <div
    className="c2"
  >
    <form
      action="#"
      className="c3"
      id="joinGameForm"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="c4"
      >
        <label
          className="sc-fznxsB c5"
          htmlFor="name"
        >
          Enter Your Name
        </label>
        <input
          className="c6"
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          tabIndex="1"
          type="text"
          value=""
        />
      </div>
      <div
        className="c4"
      >
        <label
          className="sc-fznxsB c5"
          htmlFor="gamecode"
        >
          Enter a Game Code
        </label>
        <input
          className="c6"
          id="gamecode"
          name="gamecode"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          tabIndex="1"
          type="text"
          value=""
        />
      </div>
    </form>
    <p
      className="c7"
    >
      Joining game 
    </p>
  </div>
  <div
    className="c8"
  >
    <button
      className="c9"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="c10"
      disabled={true}
      form="joinGameForm"
      type="submit"
    >
      Join
    </button>
  </div>
</div>
`;
